src/slicer.cpp

```diff
@@ -671,20 +710,53 @@ Slicer::Slicer(Mesh* mesh, int initial, int thickness, int layer_count, bool kee
             if (layer_nr < 0) continue;
             
             SlicerSegment s;
+            int end_edge_start_pt = -1;
             if (p0.z < z && p1.z >= z && p2.z >= z)
+            {
                 s = project2D(p0, p2, p1, z);
+                end_edge_start_pt = 0;
+                if (p1.z == z)
+                {
+                    s.endOtherFaces = v1.connected_faces;
```

Comment:
Wouldn't it be better to just store an (optional) pointer to the vertex, rather than copying a vector?

The check at line 88 would then be `segment.end_vertex == nullptr` rather than `segment.endOtherFaces.empty()`

Also note that a segment could end in a vertex of a face which is not properly connected to other faces in case of a bad model.